jerry-core/ecma/builtin-objects/ecma-builtin-math.c

```diff
@@ -126,10 +127,12 @@ ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
       ecma_fast_free_value (value);
     }
 
-    if (JERRY_UNLIKELY (ecma_number_is_nan (arg_num)))
+    if (JERRY_UNLIKELY (nan_found || ecma_number_is_nan (arg_num)))
     {
-      result_num = arg_num;
-      break;
+      nan_found = true;
+      arg++;
```

Comment:
As i see the 
```
arg++;
args_number--;
```
can be performed before the `isnan` and `iszero` conditions so I'd move them up, therefore no duplication is required.

Reply:
Good point, fixed.